test/unittests/api_test.cpp

```diff
@@ -315,22 +315,22 @@ TEST(api, find_exported_global)
 
     auto opt_global = find_exported_global(*instance, "g1");
     ASSERT_TRUE(opt_global);
-    EXPECT_EQ(*opt_global->value, 0);
+    EXPECT_EQ(opt_global->value->as<uint32_t>(), 0);
```

Comment:
You change the test here to check only 32-bit, I thinks it's better to check `.i64`

Reply:
Changed it to i32 because that's what the sources use above.

Reply:
In case `find_exported_global` would return some incorrect `i64` global with the equal lower word, we would miss it.

Reply:
I'm for staying with `i64` just because it tests more bits and goes along with the current implementation.

Reply:
With this reasoning, are we also using `.i64` for inputs even if the function type is defined as i32?

Reply:
> With this reasoning, are we also using `.i64` for inputs even if the function type is defined as i32?

Not for computation, but you can add asserts there.

Reply:
That is a tad inconsistent. Perhaps the best really is a helper which asserts there are no dirty higher bits.

Reply:
Testing more than is defined by wasm type system is fine to me.

Reply:
Solved this by adding the helper.